filebeat/input/filestream/config.go

```diff
@@ -237,9 +252,17 @@ func (c config) checkUnsupportedParams(logger *logp.Logger) {
 				"duplication and incomplete input metrics, it's use is " +
 				"highly discouraged.")
 	}
-	if c.GZIPExperimental {
-		logger.Named("filestream").Warn(cfgwarn.Beta(
-			"filestream: beta gzip support enabled"))
+	if c.GZIPExperimental != nil {
+		logger.Named("filestream").Warn(cfgwarn.Deprecate(
+			"",
+			"'gzip_experimental' is deprecated and ignored, set 'compression' instead"))
+	}
+
+	// file_identity should be fingerprint when compression is enabled.
+	if (c.Compression == CompressionGZIP || c.Compression == CompressionAuto) &&
+		(c.FileIdentity != nil && c.FileIdentity.Name() != fingerprintName) {
+		logger.Warnf(
```

Comment:
Now that the compression setting is opt in, I am softening up on my requirement that this shouldn't be an error. Thinking more this feels like it will be like the ID not being required eventually where we are constantly explaining it.

I was worried that when our default was implicitly auto, people would accidentally ingest gzip with the wrong file identity and cause Filebeat to exit. I don't think this risk exists anymore.

Let's make not using the fingerprint identity an error here.

Reply:
@orestisfl, could you validate I made the changes Craig asked? they're here: https://github.com/elastic/beats/pull/47893/changes/f6698a9d389b95ca695a9ba168389a10a7b3fc90